subsys/net/l2/wifi/wifi_utils.c

```diff
@@ -223,13 +247,20 @@ int wifi_utils_parse_scan_bands(char *scan_bands_str, uint8_t *band_map)
 		return -EINVAL;
 	}
 
-	band_str = strtok_r(scan_bands_str, ",", &ctx);
+	parse_str = wifi_utils_strdup(scan_bands_str);
+
```

Comment:
BTW, we usually do not place empty line after setting a var and then checking its value (because they kind of belong together).